init.lua

```diff
@@ -0,0 +1,572 @@
+  return
+end
+
+---}}}
+
+---{{{ Colorscheme
+
+balls.register({
+  url = "https://github.com/catppuccin/nvim.git",
+  name = "catppuccin",
+})
+
+local catppuccin_installed, catppuccin = pcall(require, "catppuccin")
+
+if catppuccin_installed then
+  -- See `:help catppuccin-configuration` if you want to do any more configuration.
+  catppuccin.setup({
+    flavour = "mocha",
+  })
+
+  -- neovim also comes with a bunch of builtin colorschemes. Run `:colorscheme ` and hit tab!
+  -- Most of them are pretty hideous but `habamax` and `quiet` are pretty good.
+  --
+  -- Catppuccin is better though :)
+  vim.cmd.colorscheme("catppuccin")
+end
+
+--}}}
+
+---{{{ Treesitter
+---
+--- Treesitter is responsible for better syntax highlighting, indentation, folding, and more!
+--- The way it is setup it will install any missing parsers for the current file automatically.
+--- If you want to explicitly install a parser run `:TSInstall <language>` or put it in the
+--- `ensure_installed` table below.
+---
+--- Treesitter requires a C compiler to compile the parsers. If you're on Windows you might not have
+--- a C compiler installed, or the one you have doesn't work very well with treesitter. From
+--- experience zig has proven to be the most reliable on Windows, so if you have issues with
+--- treesitter I recommend you install zig:
+---
+---   <https://ziglang.org/download/>
+---
+--- And read this section of the nvim-treesitter docs:
+---
+---   <https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support#which-c-compiler-will-be-used>
+---
+--- And tell treesitter to use zig for compiling parsers.
+---
+--- To make sure everything goes smoothly I also recommend deleting old failed parsers before trying
+--- to recompile them with zig. To do that, simply delete the following directory starting at the
+--- root of your nvim config: `pack/balls/start/nvim-treesitter/parser`
+---
+--- Then try to compile the parsers again.
+
+balls.register({ url = "https://github.com/nvim-treesitter/nvim-treesitter.git" })
+
+local treesitter_installed, treesitter = pcall(require, "nvim-treesitter.configs")
+
+if treesitter_installed then
+  -- See `:help nvim-treesitter-quickstart` for more information.
+  treesitter.setup({
+    -- You can list parser you definitely want installed here.
+    -- Most of these are recommended because they are used in documentation such as `:help` so you
+    -- might get errors if you don't install them. `vimdoc`, `bash` and `python` for example are all
+    -- used in `:help`, so you should have them installed.
+    --
+    -- You can add whatever other languages you regularly work with here as well. For a full list of
+    -- supported languages refer to this:
+    --
+    --   <https://github.com/nvim-treesitter/nvim-treesitter/#supported-languages>
+    --
+    -- I added Rust, JavaScript and TypeScript as examples, but feel free to put whatever languages
+    -- you want in here :)
```

Comment:
Most of this is mentioned under `ensure_installed` in a probably more concise and straight forward manner, could remove this. keep the middle part with the link tho